packages/experimental/Button/SPEC.md

```diff
@@ -0,0 +1,191 @@
+# Button
+
+## Background
+
+The `Button` component enables users to trigger an action or event, such as submitting a form, opening a dialog, canceling an action, or performing a delete operation.
+
+## Sample Code
+
+Basic examples:
+
+```jsx
+<Button>Text</Button>
+<Button icon={<SVGIcon />} />
+<Button icon={<SVGIcon />}>Text</Button>
+<Button icon={<SVGIcon />} iconPosition="after">Text</Button>
+<Button appearance="primary">Text</Button>
+<Button disabled>Text</Button>
+<Button size="small">Text</Button>
+<Button size="large">Text</Button>
+```
+
+More examples in the Tester App.
+
+## Variants
+
+### Appearance
+
+The `Button` component has several apparance variants depending on where it's being used:
+
+- The default `Button` is rendered with its default styling indicating a trigger for an action.
+- appearance="primary": The `Button` is styled to emphasize that it represents the primary action.
+- appearance="subtle": The `Button` is styled to blend into its background to become less emphasized.
+- appearance="transparent": The `Button` is styled such that it has no background or border styling and is just emphasized through its content styling.
+
+### Icon
+
+The `Button` component can include an `icon` that appears before or after its `children`. If an `icon` is provided without any `children`, then the `Button` becomes an icon-only `Button`.
+
+### Shape
+
+- shape="rounded": The button as rounded corners. This is the default is shape is not set.
+- shape="circular": The button has completely round corners. A button of equal width and height will be a circle.
+- shape="square": The button has right-angle corners.
+
+### Sizes
+
+The `Button` component supports different sizing with at least three different sizes: `small`, `medium` (default) and `large`.
+
+### Block
+
+The `Button` component can completely fill the width of its container.
+
+### Loading
+
+The `Button` component can be loading if it's waiting for another action to occur before allowing itself to be interacted with.
```

Comment:
As in, it would have an ActivityIndicator inside of it?

Reply:
I think? @ValentinaKozlova might have more context from design. I haven't seen redlines for this yet.

Reply:
It's a good question. I asked the Design

Reply:
I talked to Design. As I understood ActivityIndicator is mostly for other controls. But we also can use it in Button component.

Reply:
There is a (very new, in the last year or so) precedent for ActivityIndicators inside buttons on iOS is why I asked. Previously, I would have thought that a strange design, till apple made it one of their officially supported button configurations 🤷 (https://developer.apple.com/documentation/uikit/uibutton/configuration/3750795-showsactivityindicator?changes=l_8)

I suppose how we actually achieve the loading effect is a detail, but thanks for satisfying my curiosity :)